docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,165 @@
+|[runtime](#runtime-assets)|Managed assemblies copied to the output directory. `runtimes/{rid}/lib/{tfm}/` if it exists, otherwise `lib/{tfm}/`.|
+|[native](#native-assets)|Native libraries copied to the output directory. `runtimes/{rid}/native/`.|
+
+In all cases, NuGet chooses the "most compatible" directory for a given RID and Target Framework, and only the files in that directory are copied.
+Therefore, you cannot put common assets in an inherited RID (such as `any`), because if any other RID is a better match, the `any` RID contents will not be considered.
+
+To inspect the implementation of NuGet's asset selection, see [`NuGet.Client`'s `ManagedCodeConventions.cs` file](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs).
+
+### Compile assets
+
+Compile assets are passed to the compiler, and therefore must only contain .NET assemblies.
+NuGet will select compile assets from `ref/{tfm}/`, and if that directory does not exist, it will fall back to `lib/{tfm}`, where `{tfm}` is the target framework that is the closest match the project referencing the package.
+However, in this scenario of creating a .NET package with native libraries, it is recommended to use `ref` because if `lib/` is used, NuGet will think the package is compatible with projects using `packages.config`, but those projects will fail at runtime because the native files will not be copied.
+
+The .NET compiler will generate warnings when an assembly targets a different architecture than what the current compilation is targeting, so your package consumers will have the best experience if your `ref/` or `lib/` assemblies target `AnyCPU`.
+
+If your assembly contains a lot of managed code, not just P/Invoke method definitions, you can save space by making the `ref/` assembly a [reference assembly](/dotnet/standard/assembly/reference-assemblies#generating-reference-assemblies), but be sure to include it under `ref/` in your package, not `lib/`.
+
+Therefore, there is no capability to provide different compile-time assemblies for different RIDs, you will need to provide the same API surface for all RIDs, and do runtime checks (for example, throw `PlatformNotSupportedException`).
+
+### Runtime assets
+
+Runtime assets are .NET managed assemblies copied to the output directory on build and publish, and the .NET SDK generates the `deps.json` with all the RID-specific asset information.
+NuGet will select runtime assets from `runtimes/{rid}/lib/{tfm}/`, and if that doesn't exist, then it will fall back to `lib/{tfm}/`.
+The best `{rid}` directory is selected first, followed by `{tfm}`.
+However, in this scenario of creating a .NET package with native libraries, it is recommended to use `runtimes/` because if `lib/` is used, NuGet will think the package is compatible with projects using `packages.config`, but those projects will fail at runtime because the native assets will not be copied.
```

Comment:
This seems like one of those really painful "gotchas". If I'm not mistaken, we get customer feedback every so often that's related to this. The whole, "Where's my published assets" question that @jeffkl recently helped me understand. Jeff may have something to say about this section.

Reply:
I'm glad this information seems useful. I'm resolving this conversation due to lack of actionable changes to the PR.